chia/cmds/coin_funcs.py

```diff
@@ -36,6 +27,15 @@ async def async_list(args: Dict[str, Any], wallet_client: WalletRpcClient, finge
     if not await wallet_client.get_synced():
         print("Wallet not synced. Please wait.")
         return
+    min_coin_amount = Decimal(args["min_coin_amount"])
+    max_coin_amount = Decimal(args["max_coin_amount"])
+    excluded_coin_ids = args["excluded_coin_ids"]
+    excluded_amounts = args["excluded_amounts"]
+    addr_prefix = args["addr_prefix"]
+    show_unconfirmed = args["show_unconfirmed"]
+    paginate = args["paginate"]
+    if paginate is None:
+        paginate = sys.stdout.isatty()
```

Comment:
what's the reason to move this until after checking the wallet type?
Is it to fail earlier if we have the wrong wallet ID?

Reply:
yep, also looks cleaner

Reply:
> given that we don't have any test coverage for this, it seems risky to move things around like this. It's not trivial to manually inspect to ensure it's correct. (e.g. moved verbatim and no values mutate in the collapsed section of the patch that changes meanings of variables from one place to the other)

if you view commit by commit its really easy to see this change.

Reply:
it's hard to review commit-by-commit because later amendments weren't squashed into the original commits